src/definitions/mso.rs

```diff
@@ -63,6 +63,9 @@ pub struct Mso {
 
     /// Information about the validity of the Mso object.
     pub validity_info: ValidityInfo,
+
+    #[serde(flatten)]
+    pub extra_fields: BTreeMap<String, ciborium::Value>
```

Comment:
According to ISO/IEC 18013-5 extra fields in the MSO are not defined. ISO/IEC 23220-4 introduces a new field, but still does not have a provision for arbitrary extra fields. I'm hesitant to add any potential footguns that could affect interoperability.

Can I ask what the use case is for this?

Reply:
Some auxiliary standards like IETF Token Status List introduce additional fields into the MSO object ([section 6.3](https://www.ietf.org/archive/id/draft-ietf-oauth-status-list-06.html#section-6.3))

Reply:
Are there currently any mDL implementations that use the IETF Token Status List?

I'm not particularly comfortable to update the ISO-defined data structures to meet the requirements of external draft specifications, as that could easily lead to confusion and misuse. I'd rather define a way to extend the definitions during verification of the MSO.

Using serde it would be trivial to define a new struct that contains the MSO and additional fields in your own codebase, you would just to update this library to provide access to the `issuerAuth` object during verification.

Reply:
At the moment, I am not aware about other implementations of IETF TSL.

Practically speaking, this `extra_fields` implementation is similar to how `ssi` allows its users to add customized fields into standardized structures.

Reply:
The API and protocol for mdocs are significantly different than VCs in ssi. This library doesn't force you to use the `Mso` struct that is defined here. You can just as easily deserialize the payload of IssuerAuth into an `IetfMso` structure that you define.